hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -900,6 +901,173 @@
       tests: []
     }
 
+    // SYSRST_CTRL (pre-verified IP) integration tests:
+    {
+      name: chip_sysrst_ctrl_inputs
+      desc: '''Verify that the sysrst_ctrl input pin values can be read.
+
+            - Drive a known value on ac_reset, ec_rst_l, pwrb and key* pins at the chip IOs.
+            - Read the pin_in_value CSR to check for correctness.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_sysrst_ctrl_outputs
+      desc: '''Verify that the sysrst_ctrl output pin values can be set.
+
+            - Drive a known value on ac_reset, ec_rst_l, pwrb and key* pins at the chip IO inputs.
+            - Verify that sysrst ctrl correctly loops them back to the chip IO outputs.
+            - Write the pin allowed ctl register to allow some of the pins to be overridden with
+              either 0 or 1 or both.
+            - Write the pin out ctrl register to enable the override on some of the pins.
+            - Write the pin out values register to set a pattern on those pins.
+            - Verify that at the chip IO outputs, pins on which override should be active is
+              reflecting the overridden values. All others should reflect the values driven on chip
+              inputs.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_sysrst_ctrl_in_irq
+      desc: '''Verify the SYSRST ctrl can detect an input combination to signal in interrupt.
+
+            - Program a specific combination of transitions on pwrb, key*, ac_present and ec_reset_l
+              pins to trigger an interrupt by writing to key_intr_ctl register.
+            - Program the key_intr_debounce_ctl register to debounce an appropriate time.
+            - Enable the interrupt at sysrst ctrl as well as at the PLIC.
+            - Glitch the inputs at the chip IOs before stabilizing on the programmed transitions.
+            - SW services the interrupt when triggered, verifies the pin input value and
+              key_intr_status for correctness and clears the interrupt status.
+            - Verify separately, eack key combination sufccessfully generates an interrupt.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_sysrst_ctrl_sleep_gsc_wakeup
+      desc: '''Verify the SYSRST ctrl can wake up the chip from deep sleep.
+
+            - Read the reset cause register in rstmgr to confirm we are in POR reset phase.
```

Comment:
Why depend on coming from POR reset? This test could be run after some other test...